crates/pg_completions/src/providers/functions.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{
+    builder::CompletionBuilder, context::CompletionContext, relevance::CompletionRelevanceData,
+    CompletionItem, CompletionItemKind,
+};
+
+pub fn complete_functions(ctx: &CompletionContext, builder: &mut CompletionBuilder) {
+    let available_functions = &ctx.schema_cache.functions;
+
+    let completion_items: Vec<CompletionItem> = available_functions
+        .iter()
+        .map(|foo| CompletionItem {
```

Comment:
Nit: can't we loop over available functions and add the item dirextly?

Reply:
great point, saves one full iteration – adjusted